### ISSUE TITLE
[BUG] The bufferline sometimes cannot work well with offsets

### ISSUE THREAD

**Issue by nullptr-yxw** [akinsho/bufferline.nvim#798]
When I open the neovim by directory, there is a line like this, even though I have set options.offset.test = "File Explorer"
![image](https://github.com/akinsho/bufferline.nvim/assets/96286337/8cfe1f35-488c-47e7-bfaf-9b45c1515db2)
And it will disappear when I move my cursor by mouse or keyboard

https://github.com/akinsho/bufferline.nvim/assets/96286337/60943e71-23cc-4215-8253-4668a3395e58

Part of my settings is
```lua
offsets = {
    {
        filetype = "NvimTree",
        text = "File Explorer",
        highlight = "Directory",
        text_align = "left",
        separator = true,
    },
},
always_show_bufferline = true,
```
So is it possible to make the text of that line be "File Explorer", and do not disappear while moving the cursor?
Just like this(barbar.nvim)

https://github.com/akinsho/bufferline.nvim/assets/96286337/c5b082c1-6637-41a9-be68-5a2a8479e836

And there is another problem I have met which might be associated to this.
Sometimes when I close a buffer, the bufferline will not respect the offset

https://github.com/akinsho/bufferline.nvim/assets/96286337/778a74a4-05d4-43f6-9036-e36266e722fa

Another bufferline plugin barbar.nvim doesn't have this problem as well

https://github.com/akinsho/bufferline.nvim/assets/96286337/53780e31-d239-492c-9e09-7188b091bf68

**Comment by akinsho** [comment]
@nullptr-yxw please don't open multiple issues if they aren't *very clearly* indicated why each is different. I'm only one person maintaining this repo with lots of new issues all the time it just makes things harder. Please just detail precisely the problem on this one issue and if possible look at the code to consider helping raise a PR as well 🙏🏾

**Comment by nullptr-yxw** [comment]
> @nullptr-yxw please don't open multiple issues if they aren't _very clearly_ indicated why each is different. I'm only one person maintaining this repo with lots of new issues all the time it just makes things harder. Please just detail precisely the problem on this one issue and if possible look at the code to consider helping raise a PR as well 🙏🏾

Ok, I have change this issue to describe the two problem in one issue

**Comment by akinsho** [comment]
@nullptr-yxw there is some confusion about this feature. Offsets are *only relevant when there is more than one window open*. When you open neovim in a directory and it is the only window open offsets do not apply, they never have that is not why they were designed.

This can be added if you want to contribute it but is not the use case I designed them for.
Regarding the naming of the offset that should work consistently and does for me but as I have mentioned offsets are not designed to apply when there is only one window open.

When there is only one window open, there is no offset at all because nothing is being offset.
Offsets are *not a perfect solution without issue* because offsets are a bit of a hack around native neovim behaviour. They do not cover all use cases and when I added the feature I explained that I was not going to sink hours of my time into fixing every edge case. The main use case for offsets is right or left aligned side bar that is open *beside another window*.

**Comment by nullptr-yxw** [comment]
> @nullptr-yxw there is some confusion about this feature. Offsets are _only relevant when there is more than one window open_. When you open neovim in a directory and it is the only window open offsets do not apply, they never have that is not why they were designed.
> 
> This can be added if you want to contribute it but is not the use case I designed them for. Regarding the naming of the offset that should work consistently and does for me but as I have mentioned offsets are not designed to apply when there is only one window open.
> 
> When there is only one window open, there is no offset at all because nothing is being offset. Offsets are _not a perfect solution without issue_ because offsets are a bit of a hack around native neovim behaviour. They do not cover all use cases and when I added the feature I explained that I was not going to sink hours of my time into fixing every edge case. The main use case for offsets is right or left aligned side bar that is open _beside another window_.

But what about the second problem? I only closed the main.cpp, and the functor.cpp still exists, but the behavior is not normal as well

**Comment by akinsho** [comment]
@nullptr-yxw the issue you are seeing is because bufferline uses the *the native mechanism* to close buffers in neovim `:bdelete` this command does not preserve your windows and is a *well known and often discussed* issue in the (n)vim ecosystem. There is a config option to replace that default command with some command of your own liking *please read the README for how to configure the close action*. Then use some plugin like `nvim-bufdel` or `bufdelete.nvim` etc. there are loads of options for that

**Comment by nullptr-yxw** [comment]
> @nullptr-yxw the issue you are seeing is because bufferline uses the _the native mechanism_ to close buffers in neovim `:bdelete` this command does not preserve your windows and is a _well known and often discussed_ issue in the (n)vim ecosystem. There is a config option to replace that default command with some command of your own liking _please read the README for how to configure the close action_. Then use some plugin like `nvim-bufdel` or `bufdelete.nvim` etc. there are loads of options for that

Thanks for your answer and advice very much.

**Comment by akinsho** [comment]
The first issue *might* end up being fixed by #800 but in either case closing this out
